spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarAdministration.java

```diff
@@ -161,39 +164,52 @@ private void createOrModifyTopicsIfNeeded(Collection<PulsarTopic> topics) {
 	}
 
 	private void doCreateOrModifyTopicsIfNeeded(PulsarAdmin admin, Collection<PulsarTopic> topics) {
-		Map<String, List<PulsarTopic>> topicsPerNamespace = getTopicsPerNamespace(topics);
+		var topicsPerNamespace = getTopicsPerNamespace(topics);
 
 		topicsPerNamespace.forEach((namespace, requestedTopics) -> {
-			Set<PulsarTopic> topicsToCreate = new HashSet<>();
-			Set<PulsarTopic> topicsToModify = new HashSet<>();
+			var topicsToCreate = new HashSet<PulsarTopic>();
+			var topicsToModify = new HashSet<PulsarTopic>();
 
 			try {
-				List<String> existingTopicsInNamespace = admin.topics().getList(namespace);
+				var existingTopicsInNamespace = admin.topics().getList(namespace);
 
-				for (PulsarTopic topic : requestedTopics) {
+				for (var topic : requestedTopics) {
+					var topicName = topic.getFullyQualifiedTopicName();
 					if (topic.isPartitioned()) {
-						List<String> matchingPartitions = getMatchingTopicPartitions(topic, existingTopicsInNamespace);
+						if (existingTopicsInNamespace.contains(topicName)) {
```

Comment:
This is part one of the change: "If there is already a non-partitioned topic out there - barf"